subprojects/testing-native/src/test/groovy/org/gradle/nativeplatform/test/xctest/internal/XCTestSelectionTest.groovy

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.test.xctest.internal
+
+import org.gradle.testing.internal.util.Specification
+import spock.lang.Subject
+
+@Subject(XCTestSelection)
+class XCTestSelectionTest extends Specification {
+    def "includes all tests when no filter provided"() {
+        expect:
+        select([]).includedTests == [XCTestSelection.INCLUDE_ALL_TESTS]
+    }
+
+    def "throws IllegalArgumentException when filter has more than two dots"() {
+        when:
+        select(['more.than.two.dots'])
+
+        then:
+        def ex = thrown(IllegalArgumentException)
+        ex.message == 'Invalid pattern'
+    }
+
+    def "can use filters with two dots or fewer"() {
+        when:
+        select(['one.dot', 'has.two.dots'])
+
+        then:
+        noExceptionThrown()
+    }
+
+    def "converts second dot into slash"() {
+        expect:
+        select(['one.dot', 'has.two.dots']).includedTests == ['one.dot', 'has.two/dots']
+    }
+
+    def "can use wildcard to select all tests from module"() {
+        expect:
+        select(['ModuleName.*']).includedTests == ['All']
+    }
+
+    def "can use wildcard to select all test case from suite"() {
+        expect:
+        select(['ModuleName.suite.*']).includedTests == ['ModuleName.suite']
+    }
+
+    def "ignores textual duplicate filters"() {
+        expect:
+        select(['a.b.c', 'a.b.c']).includedTests == ['a.b/c']
+    }
+
+    def "ignores conceptual duplicate filters"() {
+        expect:
+        select(['a.*', 'a.b.c']).includedTests == ['All']
+        select(['a.b.*', 'a.b.c', 'a.d.e']).includedTests == ['a.b', 'a.d/e']
+        select(['a.b.c', 'a.d.e', 'a.b.*']).includedTests == ['a.d/e', 'a.b']
+    }
+
+    def "conserve order of filters"() {
+        expect:
+        select(['a.1', 'a.2', 'a.3', 'a.4']).includedTests == ['a.1', 'a.2', 'a.3', 'a.4']
+    }
+
+    private static XCTestSelection select(List<String> commandLinePattern) {
```

Comment:
nit, you could also do `select(String... commandLinePatterns)` and drop all of the `[]`'s

Reply:
Make sense, let me fix this.